hawkrest/__init__.py

```diff
@@ -23,10 +23,44 @@
 default_message_expiration = 60
 
 
+def lookup_credentials(cr_id):
+    if cr_id not in settings.HAWK_CREDENTIALS:
+        raise LookupError('No Hawk ID of {id}'.format(id=cr_id))
+    return settings.HAWK_CREDENTIALS[cr_id]
+
+
+def lookup_user(request, credentials):
+    return HawkAuthenticatedUser(), None
+
+
+settings_user_lookup = getattr(
+    settings,
+    'HAWK_USER_LOOKUP',
+    None)
+user_lookup = (
+    import_string(settings_user_lookup) if settings_user_lookup
+    else lookup_user)
+
+settings_cred_lookup = getattr(
+    settings,
+    'HAWK_CREDENTIALS_LOOKUP',
+    None)
+credentials_lookup = (
+    import_string(settings_cred_lookup) if settings_cred_lookup
```

Comment:
I'm sort of trying to have it both ways: keeping the current override mechanism (via settings) and extending that to the user lookup, retaining compatibility with current deployments, and also allowing for easy subclassing (by overriding `_credentials_lookup()` and/or `_user_lookup()`. I also wanted to have them as methods, so a subclass can define `__init__()` as desired, say with instance-specific credentials list, or what have you. I did consider assigning them as attributes in `__init__()` on the base class, but subclassing is always easier with a clean `__init__()`, IMO.